Makefile

```diff
@@ -1,51 +1,77 @@
-OS_NAME := $(shell uname -s | tr A-Z a-z)
-EXEC = bin/LRez
-ifeq ($(OS_NAME),linux)
-        SHLIB_EXT=so
-else
-        SHLIB_EXT=dylib
+ifeq ($(SHLIB_EXT),)
+	OS_NAME := $(shell uname -s | tr A-Z a-z)
+	ifeq ($(OS_NAME),linux)
+		SHLIB_EXT=.so
+	else
+		SHLIB_EXT=.dylib
+	endif
 endif
 
 curDir = $(shell pwd)
-CC = g++
-CFLAGS  = -Wall -pedantic -O3 -m64 -shared -Wl,--whole-archive -lpthread -Wl,--no-whole-archive -std=c++11 -fPIC
 
-BAMTOOLS_INC = $(curDir)/bamtools/include/bamtools/
-BAMTOOLS_LIB = $(curDir)/bamtools/lib/
+PREFIX ?= /usr/local
+BINDIR = $(PREFIX)/bin
+LIBDIR = $(PREFIX)/lib
+
+CXX ?= g++
+CXXFLAGS += -Wall -pedantic -O3 -m64 -std=c++11 -fPIC
+LDFLAGS += -L./lib -Wl,-rpath,$(LIBDIR) -Wl,--whole-archive -lpthread -Wl,--no-whole-archive
+
+BAMTOOLS_LIB_PREFIX = lrez_
+BAMTOOLS_LIB = lib/lib$(BAMTOOLS_LIB_PREFIX)bamtools$(SHLIB_EXT)
 
-LREZ_INC = $(curDir)/src/include/
-LREZ_LIB = $(curDir)/lib/
+BAMTOOLS_INC = ./include/bamtools/
+LREZ_INC = ./src/include/
 
-LDFLAGS_BAMTOOLS = -lbamtools -L$(BAMTOOLS_LIB)
-LDFLAGS_LREZ = -llrez -L$(LREZ_LIB)
-LDFLAGS_BOOST_LZ_LM = -lboost_iostreams -lz -lm -lc
+LIBS_LREZ = -llrez
+LIBS_BOOST_LZ_LM = -lboost_iostreams -lz -lm -lc
+LIBS_BAMTOOLS = -l$(BAMTOOLS_LIB_PREFIX)bamtools
```

Comment:
It's common to split `LDFLAGS` and `LIBS` since one can then more easily control their positioning (some flags introduce positional behavior even nowadays).